src/deploy/functions/release/fabricator.ts

```diff
@@ -45,7 +45,7 @@ export interface FabricatorArgs {
   appEngineLocation: string;
 
   // Required if creating or updating any GCFv1 functions
-  sourceUrl?: string;
+  sourceUrls?: Record<string, string>;
```

Comment:
We can probably just stick with one region. We had to use multiple regions in GCFv2 because GCF threatened to block cross-region installs. In GCFv1 I don't think they can make that change retroactively. And we're not violating data residency if we know they'll need to store functions in that region anyway.

Reply:
But which region given the possibility of multiple regions on a function source (ties nicely w/ the analytics PR update you made earlier)?

Reply:
The analytics PR tracks the regions we deploy to, not the regions we upload to. We _can_ make this multi-regional if you really believe in it, but it seems like

A) We're not going to run into data residency problems since the dev clearly has access to that region
B) We're going to reduce bandwidth from the user to not upload X MB to Y regions whereas we used to only upload X MB

If you really believe in per-region uploads though I won't stand in your way.

Reply:
I thought we had no choice but to have per-region uploads, but I see now that I had unnecessary strict requirements for myself. Happy to not go down that path unless we change our minds later.